include/fastrtps/rtps/builtin/discovery/endpoint/EDPSimple.h

```diff
@@ -146,20 +182,13 @@ class EDPSimple : public EDP
                 const ReaderProxyData& remote_reader_data) override;
 #endif
 
+    std::mutex temp_data_lock_;
+    ReaderProxyData temp_reader_proxy_data_;
+    WriterProxyData temp_writer_proxy_data_;
 };
 
-
-// Default configuration values for EDP entities.
- extern const Duration_t edp_heartbeat_period;
- extern const Duration_t edp_nack_response_delay;
- extern const Duration_t edp_nack_supression_duration;
- extern const Duration_t edp_heartbeat_response_delay;
```

Comment:
Default EDP times declarations are moved from EDPSimple.h to EDPServer.h.
If access to these members is no longer direct (but through new virtual APIs) then it doesn't make any sense to declare them in any header. They should be declared in EDPSimple.cpp (base class of EDPServer) as static (not reachable out of the compilation unit).

Reply:
Done on 3855427

Reply:
success!!!! 🎉